src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/CDKExhaustiveFragmenter.java

```diff
@@ -0,0 +1,283 @@
+
+import javafx.beans.property.Property;
+import javafx.beans.property.SimpleIntegerProperty;
+
+import org.openscience.cdk.DefaultChemObjectBuilder;
+import org.openscience.cdk.fragment.ExhaustiveFragmenter;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.smiles.SmilesParser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Wrapper class that makes the
+ * <a href="https://cdk.github.io/cdk/latest/docs/api/org/openscience/cdk/fragment/ExhaustiveFragmenter.html">
+ *     exhaustive fragmentation
+ * </a>
+ * from the CDK, available for MORTAR.
+ *
+ * @author Tom Weiß
+ * @version 1.0.0.0
+ */
+public class CDKExhaustiveFragmenter implements IMoleculeFragmenter {
+    //<editor-fold desc="Private static final variables">
+    /**
+     * The default value for the minimum fragment size used for the fragmentation.
+     */
+    private static final int DEFAULT_MINIMUM_FRAGMENT_SIZE = 6;
+    //
+    /**
+     * The name of the algorithm used for fragmentation.
+     */
+    private static final String ALGORITHM_NAME = "Exhaustive Fragmenter";
+    //</editor-fold>
+    //
+    //<editor-fold desc="Private final variables">
+    /**
+     * The minimum size of the returned fragments. This size consists of all atoms, that are connected by more than
+     * a single bond or have more than one single bond.
+     */
+    private final SimpleIntegerProperty minimumFragmentSize;
+    //
+    /**
+     * All settings of this fragmenter, encapsulated in JavaFX properties for binding in GUI.
+     */
+    private final List<Property<?>> settings;
+    //
+    /**
+     * Map to store pairs of {@literal <setting name, tooltip text>}.
+     */
+    private final HashMap<String, String> settingNameTooltipTextMap;
+    //
+    /**
+     * Map to store pairs of {@literal <setting name, display name>}.
+     */
+    private final HashMap<String, String> settingNameDisplayNameMap;
+    //
+    /**
+     * Instance of ExhaustiveFragmenter class to fragment a molecule.
+     */
+    private final ExhaustiveFragmenter cdkEFInstance;
+    //
+    /**
+     * Logger of this class.
+     */
+    private static final Logger LOGGER = Logger.getLogger(CDKExhaustiveFragmenter.class.getName());
+    //</editor-fold>
+    //
+    //<editor-fold desc="Constructor">
+    /**
+     * Constructor, all settings are initialised with their default values as declared in the respective public constants.
+     */
+    public CDKExhaustiveFragmenter() {
+        int tmpNumberOfSettings = 2;
+        this.settingNameTooltipTextMap = new HashMap<>(tmpNumberOfSettings,
+                BasicDefinitions.DEFAULT_HASH_COLLECTION_LOAD_FACTOR);
+        this.settingNameDisplayNameMap = new HashMap<>(tmpNumberOfSettings,
+                BasicDefinitions.DEFAULT_HASH_COLLECTION_LOAD_FACTOR);
```

Comment:
Please use CollectionUtil.calculateInitialHashCollectionCapacity() or HashMap.newHashMap().

Reply:
I see now that you only have one setting but are initialising the maps with a capacity of 2, ok. But for the future, please use the ways I indicated.